android/build.gradle

```diff
@@ -94,6 +153,23 @@ android {
             }
         }
     }
+
+    sourceSets.main {
+        java {
+            if (useNextStorage) {
+                srcDirs += 'src/kotlinPackage/java'
+            } else {
+                srcDirs += 'src/javaPackage/java'
+            }
+
+            if (!isNewArchitectureEnabled()) {
+                srcDirs += 'src/paper/java'
+            } else {
+                // This should be done by the RN gradle plugin, for some reason it sometimes doesn't work
```

Comment:
It this something we should file upstream?

Reply:
[RNGP already does it (as it was on RN 0.70)](https://github.com/facebook/react-native/blob/b3bb55357ac8b88cab889f0672f6f1c15ee5ebf5/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactPlugin.kt#L189-L191). I don't really remember whether it was causing compilation errors for me but right now, removing it doesn't seem to break anything. Possibly got fixed in the meantime, or just some weird configuration issue. I removed it from the PR for now.